src/cdk/v2/destinations/klaviyo_bulk_upload/utils.js

```diff
@@ -34,36 +56,38 @@ function generateLocationObject({
     zip,
     ip,
   };
+
+  return removeUndefinedValues(locationObject);
 }
 
 function transformSingleMessage(data, metadata) {
   const { context, traits } = data;
   const location = generateLocationObject(data);
   const { jobId } = metadata;
-  const traitsWithoutLocation = removeLocationAttributes(traits);
-  let transformedSinglePayload = {};
+  const { standardAttributes, customAttributes } = setStandardAndCustomTraits(traits);
+  const transformedSinglePayload = {
+    type: 'profile',
+    attributes: {
+      ...standardAttributes,
+      location,
+      properties: customAttributes,
+      anonymous_id: context.externalId[0].id,
```

Comment:
are we sure that external ID will always be present?

or should we use `context.externalId?.[0].id`

Reply:
same for rest

Reply:
Yes, this is coming from the identifier mapping in VDM which is mandatory to map. And since this destination doesn't allow sources other than RETL sources, so this mapping conditions cannot be skipped.